src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,174 @@
+            kElevatorDrumRadius.in(Meters),
+            kMinElevatorHeightMeters,
+            kMaxElevatorHeightMeters,
+            true,
+            0);
+    elevatorMotor.configure(
+        new SparkMaxConfig().apply(encoderCfg).apply(loopCfg),
+        ResetMode.kNoResetSafeParameters,
+        PersistMode.kNoPersistParameters);
+    if (Robot.isSimulation()) {
+      simElevatorMotor =
+          new SparkMaxSim(elevatorMotor, Constants.ElevatorSimConstants.m_elevatorGearbox);
+      elevatorMech =
+          mech.getRoot("root", 5, 5)
+              .append(
+                  new MechanismLigament2d(
+                      "Elevator Mech [0]", 1, 90, 10, new Color8Bit(Color.kPurple)));
+
+      if (widg == null) {
+        widg = Shuffleboard.getTab(getName()).add("Elevator", mech);
+      }
+    }
+  }
+
+  public Distance getElevatorHeight() {
+    return Inches.of(elevatorEncoder.getPosition());
+  }
+
+  public Command goUp() {
+    return startEnd(
+        () -> {
+          elevatorMotor.set(.5);
+        },
+        () -> elevatorMotor.set(0));
+  }
+
+  public Command goDown() {
+    return startEnd(
+        () -> {
+          elevatorMotor.set(-.5);
+        },
+        () -> elevatorMotor.set(0));
+  }
+
+  public Command changeElevation(Distance heightLevel) {
+    return runOnce(
+        () -> {
+          elevatorMotor
+              .getClosedLoopController()
+              .setReference(heightLevel.in(Inches), ControlType.kPosition);
+          SmartDashboard.putNumber("Setpoint", (heightLevel.in(Meters)/(kElevatorGearing))*(2*(Math.PI * kElevatorDrumRadius.in(Inches))));
+        });
+  }
+
+  public Command L0() {
+    return changeElevation(Inches.zero());
+  }
+
+  public Command L1() {
+    return changeElevation(Constants.ElevatorConstants.L1Height);
+  }
+
+  public Command L2() {
+    return changeElevation(Constants.ElevatorConstants.L2Height);
+  }
+
+  public Command L3() {
+    return changeElevation(Constants.ElevatorConstants.L3Height);
+  }
+
+  public Command L4() {
+    return changeElevation(Constants.ElevatorConstants.L4Height);
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    Distance simDist = Meters.zero();
+    LinearVelocity simVel = MetersPerSecond.zero();
+    for (Time i = Seconds.zero(); i.lt(Robot.period); i = i.plus(sparkPeriod)) {
+      m_elevatorSim.setInputVoltage(
+          simElevatorMotor.getBusVoltage() * simElevatorMotor.getAppliedOutput());
+      m_elevatorSim.update(sparkPeriod.in(Seconds));
+      simDist = Meters.of(m_elevatorSim.getPositionMeters());
+      simVel = MetersPerSecond.of(m_elevatorSim.getVelocityMetersPerSecond());
+      simElevatorMotor.iterate(
+          ((simVel.in(MetersPerSecond) / (2 * Math.PI * kElevatorDrumRadius.in(Meters)))
+                  * kElevatorGearing)
+              * 60,
+          RobotController.getBatteryVoltage(),
+          sparkPeriod.in(Seconds));
+    }
+    elevatorMech.setLength(1 + (simDist.in(Meters)));
```

Comment:
I recommend scaling the Mech2d ligament length up by at least a factor of 2 so that it is easier to see in the simulation

Reply:
I think the better solution here is to reduce the size of the mech2d